components/post_markdown/system_message_helpers.jsx

```diff
@@ -4,32 +4,36 @@
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {General, Posts} from 'mattermost-redux/constants';
+
 import ChannelStore from 'stores/channel_store.jsx';
 
 import {canManageMembers} from 'utils/channel_utils.jsx';
-import {Constants, PostTypes} from 'utils/constants.jsx';
 import {formatText} from 'utils/text_formatting.jsx';
 import * as Utils from 'utils/utils.jsx';
+
+import CombinedSystemMessage from 'components/post_view/combined_system_message';
 import PostAddChannelMember from 'components/post_view/post_add_channel_member';
 
-function renderUsername(value, options) {
-    return renderFormattedText(value, {...options, markdown: false});
+function renderDisplayName(value) {
+    return renderFormattedText(value, {markdown: false});
 }
 
-function renderUsernameForUserIdAndUsername(userId, username, options) {
-    const displayUsername = Utils.getDisplayNameByUserId(userId, options);
+function renderUsernameForUserIdAndUsername(userId, username) {
+    const displayUsername = Utils.getDisplayNameByUserId(userId);
     if (displayUsername && displayUsername.trim() !== '') {
-        return renderUsername(displayUsername);
+        return renderDisplayName(displayUsername);
     }
-    return renderUsername(username);
+
+    return renderDisplayName(username);
 }
 
 function renderFormattedText(value, options) {
     return <span dangerouslySetInnerHTML={{__html: formatText(value, options)}}/>;
 }
 
-function renderJoinChannelMessage(post, options) {
-    const username = renderUsernameForUserIdAndUsername(post.user_id, post.props.username, options);
+function renderJoinChannelMessage(post) {
```

Comment:
Why are we removing the options from all of these calls? We're still passing them into the functions [here](https://github.com/mattermost/mattermost-webapp/pull/1085/files#diff-5efa0068f4d81633b5dc5c0d8f6cb968R342)

Reply:
I should not touch it since it's not directly related to this PR. However, I can't see the use of `options` when it's passed at [renderUsernameForUserIdAndUsername](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_markdown/system_message_helpers.jsx#L19). Both [getDisplayNameByUserId](https://github.com/mattermost/mattermost-webapp/blob/master/utils/utils.jsx#L1056) and [renderUsername](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_markdown/system_message_helpers.jsx#L24) are not using it.
Maybe, I'll leave it as is and deal with it in separate PR?

Reply:
Oh yeah, I didn't realize how weird those were. Those options are supposed to be for passing to the Markdown renderer, so I don't know why we're passing them to `renderUsernameForUserIdAndUsername`. Cleaning that up would definitely be good